lib/rubocop/rspec/hook.rb

```diff
@@ -4,39 +4,65 @@ module RuboCop
   module RSpec
     # Wrapper for RSpec hook
     class Hook < Concept
-      STANDARDIZED_SCOPES = %i[each context suite].freeze
-      private_constant(:STANDARDIZED_SCOPES)
+      def_node_matcher :extract_metadata, <<~PATTERN
+        (block
+          {
+            (send _ _ #valid_scope? $...)
+            (send _ _ $...)
+          }
+          ...
+        )
+      PATTERN
 
       def name
         node.method_name
       end
 
       def knowable_scope?
-        return true unless scope_argument
-
-        scope_argument.sym_type?
-      end
-
-      def valid_scope?
-        STANDARDIZED_SCOPES.include?(scope)
+        scope_argument.nil? ||
+          scope_argument.sym_type? ||
+          scope_argument.hash_type?
       end
 
       def example?
         scope.equal?(:each)
       end
 
       def scope
+        return :each if scope_argument&.hash_type?
+
         case scope_name
         when nil, :each, :example then :each
         when :context, :all       then :context
         when :suite               then :suite
-        else
-          scope_name
         end
       end
 
+      def metadata
+        (extract_metadata(node) || [])
+          .map { |meta| transform_metadata(meta) }
+          .flatten
+          .inject(&:merge)
+      end
+
       private
 
+      def valid_scope?(node)
+        node&.sym_type? && Hooks::Scopes::ALL.include?(node.value)
+      end
+
+      def transform_metadata(meta)
+        if meta.sym_type?
+          { meta => true }
+        else
+          meta.pairs.map { |pair| { pair.key => transform_true(pair.value) } }
```

Comment:
This check is to be able to compare those two hooks:
```
before(:example, :special) { ... }
before(:example, special: true) { ... }
```
In the second case it's a node of a `true_type?`.

Reply:
I think it would be helpful to add that comment as an actual code comment :-)